### ISSUE TITLE
agent does not continue thread from the latest state

### ISSUE THREAD

**Issue by noamoss** [hasadna/reportit-agent#49]
(following #30) real case description, contact Saroj for further details:

> כבר הייתי בשלב "לחץ על סיום השיחה", לחצתי ואז לחצתי שוב על הפניה שבסטטוס "בטיפול" הוא לא החזיר אותי לסוף אלא לשלב מוקדם יותריותר, מצ"ב צילום.  הייתי צריכה להמשיך את השיחה כאילו קודם לא הייתי בסופה

https://admin.equality.org.il/#/report/151

![image](https://user-images.githubusercontent.com/3581741/65752689-f5729200-e115-11e9-8854-ea853534dd4f.png)

This is a real problem, not because of a practical inconvenience,  but because of the risk to lose (overwrite) saved data:

> ניסיתי שוב, הוא שוב חוזר לאותה נקודה ולא לסוף. בנוסף לא רואה את המשימות. מצרפת שבו שתי משימות שנוצרו

![image](https://user-images.githubusercontent.com/3581741/65753186-20a9b100-e117-11e9-89bc-cfb95bdad4d4.png)

**Comment by noamoss** [comment]
@akariv actually, this is a duplicate of #47. Since it case as real world report, I think it should be taken care.

**Comment by noamoss** [comment]
@akariv it might be connected to #47, which was reported by me during preparations.

**Comment by akariv** [comment]
The root cause of this issue is captured in #50 .

Depends on #50

**Comment by noamoss** [comment]
@akariv I will take care of the script level, but I think this is a scenario to throw an error when processing the YAML. I (i.e. script editors) can not know by heart all valid variable names. Would you like to open a separate issue for that, or take care of it here?

**Comment by akariv** [comment]
This is something that ofer added to his script validator that (hopefully)
he will incorporate into the build process.

**Comment by noamoss** [comment]
@akariv Ofer will add validations in the UI level (meaning, he will not let editor add a non-existing variable name) - but as you can see #50 is about a temporary/local variable, but someone might still use local variable names in a wrong name or touch the YAML manually.

 In other words, I still think we need this kind of validation also in the build phase - to prevent the site from getting into an error state just because of a mistype. don't you think?

**Comment by akariv** [comment]
The problem only happens for non-temporary variables, as only those get
sent to the server (and possibly cause an error)

**Comment by akariv** [comment]
and `housing_land_type` does not start with an underscore, so it's not a temporary/local variable by definition.
It should be easy for the validator to point out that it's an invalid variable name.
